clang/lib/CodeGen/CodeGenTypes.cpp

```diff
@@ -57,11 +57,14 @@ void CodeGenTypes::addRecordTypeName(const RecordDecl *RD,
       if (auto TemplateDecl = dyn_cast<ClassTemplateSpecializationDecl>(RD)) {
         ArrayRef<TemplateArgument> TemplateArgs =
             TemplateDecl->getTemplateArgs().asArray();
+        constexpr size_t MaxMatrixParameter = 6;
+        assert(TemplateArgs.size() <= MaxMatrixParameter &&
+               "Too many template parameters for JointMatrixINTEL type");
```

Comment:
Is there a minimum matrix parameter we can also check at the same time?

Reply:
Added, thanks!

Reply:
Assert message should be updated right? Since you added a minimum requirement as well

Reply:
Well, actually, after some consideration I realized, that there is no max or min number of template parameters, it's just a fixed number of 6.